code/modules/client/preferences/_preference.dm

```diff
@@ -542,3 +542,24 @@ GLOBAL_LIST_INIT(preference_entries_by_key, init_preference_entries_by_key())
 
 /datum/preference/toggle/is_valid(value)
 	return value == TRUE || value == FALSE
+
+
+/// A string-based preference accepting arbitrary string values entered by the user, with a maximum length.
+/datum/preference/text
+	abstract_type = /datum/preference/text
+
+	/// What is the maximum length of the value allowed in this field?
+	var/maximum_value_length = 256
+
+
+/datum/preference/text/deserialize(input, datum/preferences/preferences)
+	return STRIP_HTML_SIMPLE(input, maximum_value_length)
```

Comment:
Whether or not to strip HTML should be up to consumers, IMO, I can think of plenty of cases where I wouldn't want this. It's fine as a variable.

Reply:
Sure yeah.